apiserver/errors/errors.go

```diff
@@ -56,24 +57,24 @@ func OperationBlockedError(msg string) error {
 }
 
 var singletonErrorCodes = map[errors.ConstError]string{
-	stateerrors.ErrCannotEnterScopeYet:           params.CodeCannotEnterScopeYet,
-	stateerrors.ErrCannotEnterScope:              params.CodeCannotEnterScope,
-	stateerrors.ErrUnitHasSubordinates:           params.CodeUnitHasSubordinates,
-	stateerrors.ErrDead:                          params.CodeDead,
-	stateerrors.ErrApplicationShouldNotHaveUnits: params.CodeAppShouldNotHaveUnits,
-	jujutxn.ErrExcessiveContention:               params.CodeExcessiveContention, // TODO(dqlite): remove jujutxn.ErrExcessiveContention from api errors
-	errors.ConstError(leadership.ErrClaimDenied): params.CodeLeadershipClaimDenied,
-	errors.ConstError(lease.ErrClaimDenied):      params.CodeLeaseClaimDenied,
-	ErrBadId:                                     params.CodeNotFound,
-	ErrUnauthorized:                              params.CodeUnauthorized,
-	ErrNoCreds:                                   params.CodeNoCreds,
-	ErrLoginExpired:                              params.CodeLoginExpired,
-	ErrPerm:                                      params.CodeUnauthorized,
-	ErrNotLoggedIn:                               params.CodeUnauthorized,
-	ErrUnknownWatcher:                            params.CodeNotFound,
-	ErrStoppedWatcher:                            params.CodeStopped,
-	ErrTryAgain:                                  params.CodeTryAgain,
-	ErrActionNotAvailable:                        params.CodeActionNotAvailable,
+	errors.ConstError(relationerrors.CannotEnterScopeSubordinateNotAlive): params.CodeCannotEnterScopeYet,
+	errors.ConstError(relationerrors.CannotEnterScopeNotAlive):            params.CodeCannotEnterScope,
```

Comment:
We shouldn't have these here. Instead we should return the params error at the callsite of the call. This is going to be removed at some point - **or** at the very least it should be slimmed down.

Reply:
todo: Translate all of the errors to params versions at the facade method and not use `ServerError()`. Otherwise the data will be lost. `params.TranslateWellKnownError(err)` will be helpful too.